File: skybayoneoneone_ui/src/services/DelayNoticeService.js
import http from "../http-common"; 

class DelayNoticeService {
  getAllDelayNotices(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/delayNotice/delayNotices`, searchDTO);
  }

  get(delayNoticeId) {
    return this.getRequest(`/delayNotice/${delayNoticeId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/delayNotice?field=${matchData}`, null);
  }

  addDelayNotice(data) {
    return http.post("/delayNotice/addDelayNotice", data);
  }

  update(data) {
  	return http.post("/delayNotice/updateDelayNotice", data);
  }
  
  uploadImage(data,delayNoticeId) {
  	return http.postForm("/delayNotice/uploadImage/"+delayNoticeId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new DelayNoticeService();

File: src/main/java/com/skybayoneoneone/service/DelayNoticeService.java
package com.skybayoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skybayoneoneone.domain.DelayNotice;
import com.skybayoneoneone.dto.DelayNoticeDTO;
import com.skybayoneoneone.dto.DelayNoticeSearchDTO;
import com.skybayoneoneone.dto.DelayNoticePageDTO;
import com.skybayoneoneone.dto.DelayNoticeConvertCriteriaDTO;
import com.skybayoneoneone.service.GenericService;
import com.skybayoneoneone.dto.common.RequestDTO;
import com.skybayoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface DelayNoticeService extends GenericService<DelayNotice, Integer> {

	List<DelayNotice> findAll();

	ResultDTO addDelayNotice(DelayNoticeDTO delayNoticeDTO, RequestDTO requestDTO);

	ResultDTO updateDelayNotice(DelayNoticeDTO delayNoticeDTO, RequestDTO requestDTO);

    Page<DelayNotice> getAllDelayNotices(Pageable pageable);

    Page<DelayNotice> getAllDelayNotices(Specification<DelayNotice> spec, Pageable pageable);

	ResponseEntity<DelayNoticePageDTO> getDelayNotices(DelayNoticeSearchDTO delayNoticeSearchDTO);
	
	List<DelayNoticeDTO> convertDelayNoticesToDelayNoticeDTOs(List<DelayNotice> delayNotices, DelayNoticeConvertCriteriaDTO convertCriteria);

	DelayNoticeDTO getDelayNoticeDTOById(Integer delayNoticeId);







}





